src/middleware/sitemap/globalping.js

```diff
@@ -19,15 +19,17 @@ let probesPromise = updateProbesData();
 
 module.exports = async (ctx) => {
 	ctx.params.page = ctx.params.page.replace(/\.xml$/, '');
-	let pages = (await readDirRecursive(viewsPath + '/pages/globalping', [ '_*' ])).map(p => path.relative(viewsPath + '/pages/globalping', p).replace(/\\/g, '/').slice(0, -5));
-	let probes = await probesPromise;
-	let maxPage = Math.ceil(probes.length / 50000);
+	let pages = (await readDirRecursive(viewsPath + '/pages/globalping', [ '_*' ])).filter(p => ![ 'globalping/users.html' ].some(path => p.endsWith(path))).map(p => path.relative(viewsPath + '/pages/globalping', p).replace(/\\/g, '/').slice(0, -5));
```

Comment:
Better to prefix the filename with `_` which we use as a sign it's a dynamic page and filter automatically here.

Reply:
You now have a list of valid usernames in the sitemap module, so my idea was you make that variable exported, and use it in the server router file to set the canonical + respond with a 404 for invalid usernames (instead of the current client-side 404 fallback).

Reply:
Alright, updated to export `updateProbesData` from sitemap and use that data to show 404 page and pass the username with correct casing to the page.